contributions/MPI_IS_gaussian_process/src/gaussian_process_guider.cpp

```diff
@@ -274,7 +274,17 @@ double GaussianProcessGuider::result(double input, double SNR, double time_step,
         {
             dithering_active_ = false;
         }
-        deduceResult(time_step); // just pretend we would do dark guiding...
+        try
+        {
+            deduceResult(time_step); // just pretend we would do dark guiding...
+        }
+        catch (std::runtime_error err)

```

Comment:
minor: can think of the catch args as like method arguments, and it is a good practice to pass objects by const reference where we want reference semantics (pass by reference) and do not need copy semantics (pass by value):
```suggestion
        catch (const std::runtime_error& err)
```